### ISSUE TITLE
Drag and drop for navigation seems broken in lastest pre-release! 

### ISSUE THREAD

**Issue by Parth** [microsoft/microsoft-ui-xaml#3290]
Prompted by https://github.com/microsoft/microsoft-ui-xaml/issues/3131 issue I was asked to update to the latest pre-release.

Updating to this seems to completely disable drag and drop regardless of `CanDrop` and related settings.

**Comment by ranjeshj** [comment]
@Parth  Was this working for you in a 2.5 pre-release but then regressed in a newer one? Could you specify which release it was working in and which one it stopped ? What do you mean by drag and drop being disabled ?

**Comment by Parth** [comment]
Not sure if you looked into the issue I linked. Before the prerelease I was facing a bug regarding highlight of the navigationview. But here drag and drop is working. This video doesn't capture the tooltip I see when I drag and drop (perhaps a bug in game bar or whatever) but you can see both drag and drop working and the bug I talked about in #3131 in this video.

2.4.3: https://youtu.be/1MSW26ipDls

Now in the 2.5 prerelease (I tried them all), one of these issues is solved, but another is created: https://youtu.be/aCfrSwZzdjI

None of the code has changed, regardless of how hard I try try dragging and dropping just behaves as if all of the `CanDrag` and various other flags related to drag and drop are set to `false`.

**Comment by ranjeshj** [comment]
@Parth  would you be able to share a small repro app showing your issue ? That would really help debug. @StephenLPeters can you think of any recent changes NavView change that could have caused this ? It might be worth adding a test for this scenario.

**Comment by StephenLPeters** [comment]
@chingucoding and/or @Felix-Dev did the change to treeview's drag and drop behavior get checked in already? could that be causing this?

**Comment by StephenLPeters** [comment]
Silly me, H-Navview doesn't use tree view

**Comment by chingucoding** [comment]
I don't think that those went in yet, but even if so, NavigationView uses ItemsRepeater, not Treeview so that shouldn't make a difference.

**Comment by Felix-Dev** [comment]
@StephenLPeters Exactly as @chingucoding said :)

**Comment by Parth** [comment]
@ranjeshj I believe you could create a minimal repro with UWP sample code + add drag and drop to the `NavigationView` components.

My source is available [here](https://github.com/lockbook/lockbook/tree/master/clients/windows). More specifically [this is my xaml](https://github.com/lockbook/lockbook/blob/master/clients/windows/FileExplorer.xaml) (very small). You wouldn't have any problems running the project as is.

**Comment by Parth** [comment]
The newest prerelease (`9.23`, and all the other `2.5.*`) also breaks `muxc:NavigationView.PaneFooter`:

behavior in 2.4:

![image](https://user-images.githubusercontent.com/821972/94375480-c8f6aa80-00c8-11eb-8fbf-7fffbe9561ff.png)

behavior in the 2.5 prereleases:

![image](https://user-images.githubusercontent.com/821972/94375525-225ed980-00c9-11eb-9829-d10a772fabf4.png)

Notice the footer has been pushed down.

I think there's a serious regression to the `NavigationView` in 2.5 and someone needs to take a serious look at an example with:

+ Hierarchical navigation (with lots of elements to verify scroll behavior)
+ Drag and drop
+ Footer

The surface area of the app I've linked is really pretty low, and is a reasonable starting point for such an effort.

I would be happy to assist in the effort, but as it stands right now, there's a minor bug in 2.4 associated with this element, and a major bug in all the 2.5 releases, and it's pretty concerning, to say the least.

**Comment by Felix-Dev** [comment]
@Parth Can you open a new issue for this WinUI 2.5 specific behavior? (I am seeing this as well in case of the Settings item. With WinUI 2.5 previews, the Settings item is now no longer always visible but will be pushed out of the screen by NavigationView menu items (if there are enough visible). I don't think this change was intended but more can be discussed in its own issue on the repo.)

**Comment by StephenLPeters** [comment]
I don't think this was intended either. Lets get it a dedicated issue and we can see about fixing it for the next prerelease.

**Comment by yaichenbaum** [comment]
I am getting this same issue after updating to WinUI 2.5.

**Comment by BanCrash** [comment]
This seems to be still an issue even on 2.7.0, and also on 2.6.2 and below. I created a simple test using XAML Controls Gallery.

I made a simple test adding "CanDrag="True" on a [release that have the WinUI 2.4.0-prerelease.200414001](https://github.com/microsoft/Xaml-Controls-Gallery/releases/tag/140537), and on the last main commit, that have WinUI 2.7.0-prerelease.210816001.

On the 2.4.0 it's working as expected, while on the 2.7.0 no.

2.4.0 version, with the last NavigationView example, which is the one that I modified, is working:
![GIF 07-09-2021 18-48-19](https://user-images.githubusercontent.com/11770760/132381974-262c6b23-e384-4226-b258-c1aa208ae2c0.gif)

2.7.0 version, with the same modification on the last NavigationView example, is not working:
![GIF 07-09-2021 19-04-50](https://user-images.githubusercontent.com/11770760/132383890-0ac71746-1ddf-4ef4-8dde-ca8a2c2d6bd2.gif)

You can check those examples with that little modification on my fork: 

2.4.0 which is working: https://github.com/BanCrash/Xaml-Controls-Gallery/tree/2.4.0-prerelease.200414001

2.7.0 which is not working: https://github.com/BanCrash/Xaml-Controls-Gallery/tree/2.7.0-prerelease.210816001

NOTE: I understand that 2.7.0 is a preview version, but on stable versions of 2.5 / 2.6 is not working neither...

**Comment by yaichenbaum** [comment]
@ranjeshj @StephenLPeters it's been about a year since this issue was first reported, is it possible to include a fix in 2.7.0?

**Comment by ReneeGA2020** [comment]
It's been 2 years. Any update for this bug?

**Comment by drasticactions** [comment]
I have a potential workaround for this issue.

![image](https://user-images.githubusercontent.com/898335/213923110-c9e1cf99-794b-40b5-ace4-a8fe6a47d58e.png)

Setting "CanDrag" to the NavigationItemView alone isn't enough, I also need to set it to the underlying NavigationItemViewPresenter. Then, magically, once I did that, I could drag and drop items on the NavigationView again.

**Comment by drasticactions** [comment]
@hecksmosis You should be able to attach that to the NavigationItemViewPresenter as well. https://github.com/drasticactions/MauiFeed/blob/main/src/MauiFeed.WinUI/Views/FeedSidebarItem.cs#L237-L252

I have an example of it in my app. Basically, you would attach that the same way as you would normally to a given element. 

https://github.com/drasticactions/MauiFeed/blob/main/src/MauiFeed.WinUI/Views/FeedSidebarItem.cs#L256-L281

**Comment by JoeTomkinson** [comment]
There are a number of use cases where this bit of functionality would be useful. I understand that there may be competing priorities for the next release, but it would be great if this could be looked at with it being in the backlog for nearly 2 years.
